frontend/catalyst/debug/instruments.py

```diff
@@ -49,17 +85,62 @@ def instrumentation(session_name, filename=None, detailed=False):
         session.close()
 
 
-def instrument(fn=None, *, size_from=None, has_finegrained=False):
-    """Decorator that marks functions as targets for instrumentation. Instrumentation is only
-    performed when enabled by a session.
+def instrument(fn=None, *, size_from=None, detailed=False):
```

Comment:
So the reason a different name was chosen here is that the parameter is not an imperative (like "produce detailed output"), but rather it indicates to the instrumentation framework wether the decorated function will _have_ detailed output. So the keyword is only used to properly format the YAML output, from something like
```
    - compile:
        walltime: 199.249725
        cputime: 38.820425
        programsize: 121
```

to 

```
    - compile:
        walltime: 199.249725
        cputime: 38.820425
        programsize: 121
        finegrained:
          - <captured output that didn't come from the python instrumentation functionality>
```

I think we might want to do a bit of experimentation first to make sure the output formats (console & yaml) make sense when the `@instrument` decorator is used on user functions, since in that case they would be called from inside of an existing `@instrument`ed function.

Reply:
Will revert it!